pkg/disttask/framework/storage/nodes.go

```diff
@@ -100,14 +102,26 @@ func (mgr *TaskManager) getManagedNodesWithSession(ctx context.Context, se sessi
 	if err != nil {
 		return nil, err
 	}
-	nodeMap := make(map[string][]proto.ManagedNode, 2)
+	managedNodes := make([]proto.ManagedNode, 0)
+	liveNodes := make([]proto.ManagedNode, 0)
+	var setTiDBServiceScopeBackground bool
 	for _, node := range nodes {
-		nodeMap[node.Role] = append(nodeMap[node.Role], node)
+		if node.Role == distroleutil.TiDBServiceScopeBackground {
+			setTiDBServiceScopeBackground = true
+		}
+		if node.Dead {
+			continue
+		}
+		liveNodes = append(liveNodes, node)
+		if node.Role == distroleutil.TiDBServiceScopeBackground {
+			managedNodes = append(managedNodes, node)
+		}
 	}
-	if len(nodeMap["background"]) == 0 {
-		return nodeMap[""], nil
+	if !setTiDBServiceScopeBackground {
+		return liveNodes, nil
 	}
-	return nodeMap["background"], nil
+
```

Comment:
We can do this in 2 sql

Reply:
Can you explain in more detail? I didn't understand what you meant.

Reply:
select id, role, cpu_count from meta where role="background" and dead=false;
select id, role, cpu_count from meta where role="" and dead=false;

Reply:
lgtm to use one, cheaper(1 round trip)

node count is relatively small normally

Reply:
OK